### PR TITLE
Update configs and scripts

### PR DESCRIPTION
**Pull request by micahdbak**
I've gone through the scripts used for this server as well as the configuration files to remove lines that are generated automatically, as well as redundant parts of scripts.

I've verified that the current set of configs and scripts works on a fresh Debian 12 machine on DigitalOcean, and deployed the backend to https://api2.micahdb.com.

### CODE REVIEW

File: fresh_setup.sh [hunk 1]
```diff
@@ -1,96 +1,110 @@
...
+sudo -u csss-site git clone https://github.com/CSSS/csss-site-backend.git csss-site-backend
+
+echo "----"
+echo "configure sudo for csss-site..."
+cp csss-site-backend/config/sudoers.conf /etc/sudoers.d/csss-site
+
+echo "----"
+echo "configure nginx..."
+cp csss-site-backend/config/nginx.conf /etc/nginx/sites-available/csss-site-backend
+echo "You'll need to fill out the certbot configuration manually."
```
Comment (EarthenSky) [inline fresh_setup.sh:63]: could you link to documents describing how the config should be filled out?

Comment (micahdbak) [reply fresh_setup.sh:63]: 100%! Also, I'm working on this part right now, as certbot doesn't configure our config. It configures /etc/nginx/sites-enabled/default. So I'm going to try and play with our config and remove default so it actually pastes information into ours.

File: config/nginx.conf [hunk 2]
```diff
@@ -9,14 +9,8 @@ server {
     keepalive_timeout 5;
     client_max_body_size 1G; # Was 4G
 
-    # managed by Certbot
-    ssl_certificate /etc/letsencrypt/live/api.sfucsss.org/fullchain.pem;
```
Comment (EarthenSky) [inline config/nginx.conf:13]: I'm assuming these lines got moved somewhat that is managed by certbot automatically, could you tell me a bit more about how the new certbot setup will work?

File: fresh_setup.sh (hunk 1 above)
Comment (EarthenSky) [reply fresh_setup.sh:63]: ah! I see. Do be careful that anything automatic isn't checked into source code. Or if it is, certbot knows not to overwrite it.

File: config/nginx.conf (hunk 2 above)
Comment (micahdbak) [reply config/nginx.conf:13]: Yes; now, the preferred server configuration is listed under an HTTP server (`listen 80;`), which is nginx-valid (it doesn't use SSL), but that certbot adapts to *become* an HTTPS server (`listen 443;`). This keeps the configs on the repo super simple and leaves much of the SSL configuration to how certbot handles it :)

File: config/update_config.sh [hunk 3]
```diff
@@ -1,12 +1,28 @@
 #!/bin/bash
 
+# make sure user is root
```
Comment (EarthenSky) [inline config/update_config.sh:3]: if we ever set up an automated system for deploying changes, we'll want to be careful with automating any config changes. I like the idea of requiring root. Perhaps all config changes should always be done manually

File: config/update_config.sh [hunk 4]
```diff
@@ -1,12 +1,28 @@
...
-sudo systemctl restart csss-site
+systemd-analyze verify /home/csss-site/csss-site-backend/config/csss-site.service
+# only use new service if it is valid
+if [ $? -eq 0 ]; then
+	cp /home/csss-site/csss-site-backend/config/csss-site.service /etc/systemd/system/csss-site.service
+	systemctl restart csss-site
+fi
 
+echo "3. update sudo config"
+cp /home/csss-site/csss-site-backend/config/sudoers.conf /etc/sudoers.d/csss-site
```
Comment (EarthenSky) [inline config/update_config.sh:28]: \<3

File: config/update_config.sh (hunk 3 above)
Comment (micahdbak) [reply config/update_config.sh:3]: agree with this!

File: config/update_config.sh [hunk 5]
```diff
@@ -1,12 +1,28 @@
 #!/bin/bash
 
+# make sure user is root
+user=$(whoami)
```
[the remark this answers is not in the document]
Comment (micahdbak) [reply config/update_config.sh:4]: When the configs in the repo differ from the configs on the server. Most of these are only modifyable by the root, hence the check that the root user is running the script.

**Pull request merged**